src/modules/jcms_notifications/src/JCMSQueueService.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\jcms_notifications;
+
+use Aws\Sqs\SqsClient;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Class JCMSQueueService.
+ *
+ * @package Drupal\jcms_notifications
+ */
+class JCMSQueueService {
+
+  /**
+   * @var \Aws\Sqs\SqsClient
+   */
+  protected $sqsClient;
+
+  protected $endpoint = '';
+
+  protected $queueName = '';
+
+  protected $region = '';
+
+  /**
+   * JCMSQueueService constructor.
+   *
+   * @param \Aws\Sqs\SqsClient $sqs_client
+   */
+  public function __construct(SqsClient $sqs_client = NULL) {
+    $this->endpoint = Settings::get('jcms_sqs_endpoint');
+    $this->queueName = Settings::get('jcms_sqs_queue');
+    $this->region = Settings::get('jcms_sqs_region');
+    $config = [
+      'profile' => 'default',
+      'version' => 'latest',
+      'region' => $this->region,
+    ];
+    if (!empty($this->endpoint)) {
+      $config['endpoint'] = $this->endpoint;
+    }
+    $this->sqsClient = $sqs_client ?: new SqsClient($config);
+  }
+
+  /**
+   * Gets the article data from SQS.
+   *
+   * @return array
+   */
+  public function getArticleData() {
+    $queue = $this->sqsClient->getQueueUrl([
+      'QueueName' => $this->queueName,
+    ]);
+    $messages = [];
+    while (TRUE) {
```

Comment:
at this level I wouldn't loop as this will stay inside the while as long as it consume at least one message. It is true we may want a loop to quickly go through a large queue, but this is usually put at the highest level, in our case `jcms_notifications_cron()`